app/api/v1/mediator/ballot.py

```diff
@@ -1,221 +1,262 @@
     )
 
-    return accepted_ballot
-
 
-def index_shares_by_ballot(
-    shares: List[DecryptionShare],
-) -> Dict[BALLOT_ID, Dict[GUARDIAN_ID, DecryptionShare]]:
-    """
-    Construct a lookup by ballot ID containing the dictionary of shares needed
-    to decrypt that ballot.
-    """
-    shares_by_ballot: Dict[str, Dict[str, DecryptionShare]] = {}
-    for share in shares:
-        ballot_shares = shares_by_ballot.setdefault(share.object_id, {})
-        ballot_shares[share.guardian_id] = share
+def _get_election_parameters(
+    election_id: Optional[str], request: BaseBallotRequest
+) -> Tuple[Manifest, CiphertextElectionContext, str]:
+    # Check an election is assigned
+    if not election_id:
+        election_id = request.election_id
 
-    return shares_by_ballot
+    if not election_id:
+        raise HTTPException(
+            status_code=status.HTTP_400_BAD_REQUEST,
+            detail="specify election_id in the query parameter or request body",
+        )
 
+    # TODO: load validation from repository
+    if not request.manifest:
+        raise HTTPException(
+            status_code=status.HTTP_501_NOT_IMPLEMENTED,
+            detail="Query for Manifest Not Yet Implemented",
+        )
 
-@router.post("/submit", tags=[CAST_AND_SPOIL])
-def submit_ballot(request: AcceptBallotRequest = Body(...)) -> Any:
-    """
-    Submit ballot
-    """
-    sumbitted_ballot = save_ballot_queue(request)
-    if not sumbitted_ballot:
+    if not request.context:
         raise HTTPException(
-            status_code=500,
-            detail="Ballot failed to be submitted",
+            status_code=status.HTTP_501_NOT_IMPLEMENTED,
+            detail="Query for Context Not Yet Implemented",
         )
-    return sumbitted_ballot.to_json_object()
+
+    manifest = request.manifest
+    context = request.context
+    return manifest, context, election_id
 
 
-def save_ballot_queue(casted_ballot: Any) -> Any:
-    ballot = CiphertextBallot.from_json_object(casted_ballot.ballot)
+def _submit_ballots(
+    election_id: str,
+    ballots: List[SubmittedBallot],
+) -> SubmitBallotsResponse:
     try:
-        uri = os.environ.get("MESSAGEQUEUE_URI", "amqp://guest:guest@localhost:5672")
-        params = pika.URLParameters(uri)
-        connection = pika.BlockingConnection(params)
-        channel = connection.channel()
-        channel.queue_declare(queue="submitted-ballots")
-        channel.basic_publish(
-            exchange="",
-            routing_key="submitted-ballots",
-            body=json.dumps(ballot.to_json_object()),
-        )
-        channel.close()
-        connection.close()
-    except (pika.exceptions.ChannelError, pika.exceptions.StreamLostError):
+        with get_repository(election_id, DataCollection.SUBMITTED_BALLOT) as repository:
+            cacheable_ballots = [ballot.to_json_object() for ballot in ballots]
+            keys = repository.set(cacheable_ballots)
+            print(keys)
```

Comment:
Do we really want to print keys here or was this just for debugging?

Reply:
it was for debugging, removing